newm-tx-builder/src/main/kotlin/io/newm/txbuilder/ktx/TransactionBuilderResponseKtx.kt

```diff
@@ -0,0 +1,44 @@
+package io.newm.txbuilder.ktx
+
+import com.google.iot.cbor.CborArray
+import com.google.iot.cbor.CborInteger
+import com.google.iot.cbor.CborMap
+import com.google.iot.cbor.CborObject
+import io.newm.chain.grpc.Redeemer
+import io.newm.chain.grpc.TransactionBuilderResponse
+import io.newm.chain.grpc.exUnits
+import io.newm.chain.grpc.redeemer
+
+private val KEY_FEE: CborInteger = CborInteger.create(2)
+private val KEY_REDEEMERS: CborInteger = CborInteger.create(5)
+
+interface TxFields {
+    val fee: Long
+    val redeemers: List<Redeemer>
+    // TODO: add more fields later as needed
+}
+
+fun TransactionBuilderResponse.extractFields(): TxFields {
```

Comment:
Try extracting `totalCollateral` and using it on transactions after the first one.

Reply:
> Try extracting `totalCollateral` and using it on transactions after the first one.

Done. That worked perfectly!